src/test/modules/dish/dish.resolver.ts

```diff
@@ -45,6 +69,30 @@ export class DishResolver {
     return this.service.find<DishWhereInput>(where, orderBy, limit, offset, fields);
   }
 
+  @Authorized('dish:read')
+  @Query(() => DishesPaginated)
+  async dishesPaginated(
+    @Args() { where, orderBy, limit, offset }: DishWhereArgs,
+    @NestedFields() fields: any
+  ): Promise<DishesPaginated> {
+    const { records: dishes, total } = await this.service.findAndCount<DishWhereInput>(
+      where,
+      orderBy,
+      limit,
+      offset,
+      fields.dishes || []
```

Comment:
Once we update to nodes, this could just be `fields.nodes`.  Long-term we could expose a special service method that gives the full payload so that you don't need to findAndCount and then create the payload below.